src/quantity_api/__init__.py

```diff
@@ -0,0 +1,23 @@
+"""Quantity API."""
+
+from typing import Any, Final, Protocol, Self, runtime_checkable
+
+import optype as op
+
+type Unit = Any  # TODO: unit-api
+
+__version__: Final = "0.0.1.dev0"
+__all__ = ["__version__", "Quantity"]
+
+@runtime_checkable
+class Quantity[V, U: Unit](Protocol):
+    @property
+    def value(self) -> V: ...
+    @property
+    def unit(self) -> U: ...
+
+    ### Dunder Methods
+
+    def __eq__[B: op.CanBool](
+        self, other: Self[op.CanEq[V, B], U], /
+    ) -> B: ...
```

Comment:
are we sure that `B: op.CanBool` is correct here? Reading https://github.com/jorenham/optype/blob/27d4f5e04600041b561ef2e144a1d125d8ae7650/optype/_core/_can.py#L354-L367, `np.ndarray` doesn't implement `CanBool` right?

It seems to me like rather than something which `CanBool`, the only limitation is that it slots into the `_AnyBoolT_co` param of `op.CanEq`.

So is this sufficient / better? @jorenham @nstarman 

```suggestion
    def __eq__[B](
        self, other: Self[op.CanEq[V, B], U], /
    ) -> B: ...
```

Reply:
`ndarray.__bool__` is implemented and annotated

Reply:
Is that from your recent work? Anyway, does the point not stand that there are valid returns here which don't implement `CanBool`?

Reply:
Yes, I think it stands. Looking at [JAX](https://github.com/jax-ml/jax/blob/840192d39ad335b96b8ffb3fcb4b0267e4a9c1b1/jax/_src/array.py#L340), while it does have a `__bool__` it's mostly there to fail.

Reply:
Okay, I'll make this change. Let us know if there is something we have missed @jorenham !